### PR TITLE
Core/Player: Resolved wrong valid appearance checks for tauren/female characters.

### LINKED ISSUE #20090: [master] Error creating character
**Description:**
19dcae0dd4346304d509ac866305b737a8683048 since this commit, it is not possible to create below characters because of "Error creating character" error:
- human female
- dwarf female
- gnome female
- draenei male
- orc female
- tauren male
- tauren female
- troll female
- pandaren female

**Current behaviour:**
Creating character combinations listed above is not possible

**Expected behaviour:**
Character creation should be successful

**Steps to reproduce the problem:**
1. create one of the character listed in description and you get an error

**Branch(es):** master

**TC rev. hash/commit:**  19dcae0dd434

**TDB version:**  720.00

**Operating system:** Win 7 x64

### PR DESCRIPTION
**Pull request by Zyres**
**Changes proposed:**
-  Check valid appearance (facialHair) only for male and non tauren/dranei characters.

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [X] master

**Issues addressed:** Closes #20090

**Tests performed:** (Does it build, tested in-game, etc.)
- [X] build successfull
- [X] human female created
- [X] dwarf female created
- [X] gnome female created
- [X] draenei male created
- [X] orc female created
- [X] tauren male created
- [X] tauren female created
- [X] troll female created
- [X] pandaren female created

### CODE REVIEW

**Comment by joschiwald** [comment]
it does not fix #20090 completly and this seems to be like a hack

**Comment by Zyres** [comment]
now it fix #20090 completly.

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -476,7 +476,17 @@ bool Player::Create(ObjectGuid::LowType guidlow, WorldPackets::Character::Charac
...
+
+    // no FacialHairStyle for tauren and dranei (HairStyle used instead)
+    if (createInfo->Race != RACE_TAUREN && createInfo->Race != RACE_DRAENEI)
+    {
+        // no FacialHairStyle for females
+        if (createInfo->Sex != GENDER_FEMALE)
+            SetByteValue(PLAYER_BYTES_2, PLAYER_BYTES_2_OFFSET_FACIAL_STYLE, createInfo->FacialHairStyle);
+    }
+    else
+        SetByteValue(PLAYER_BYTES_2, PLAYER_BYTES_2_OFFSET_FACIAL_STYLE, 0);    // tauren/dranei or female = 0
```
Comment (joschiwald) [inline src/server/game/Entities/Player/Player.cpp]: where is the female equivalent like jewellery or tattoos now saved?

**Comment by Zyres** [comment]
I'm not familiar with the char sections but it seems there is something missing. I thought blizzard wants to enable jewellery, tattoos and facial hair to both genders...

**Comment by lorac2k14** [comment]
Not trying to sound stupid or be a smart --- or troll (I need clarification to definition)  I have a question, How does this fall under hack-fixes?
It looks very much legit and seems to function like Blizz's version so far through my testing but again I am still testing it looks as though the issues this was addressing is addressed so far and code is nice and clean.
Maybe I just haven't hit the right toon creation combo yet? But every known combos I knew not to work now and I am continuing to create toons so I can say every combo works or so I can point out an issue still existing to help pinpoint where the issue still exists or not
 @Zyres  Nicely done **Thank you** for coming up with a solution and I hope the powers at Trinitycore do implement this code.

**Comment by AussieGlenn** [comment]
Hi,

Hope this gets committed so female characters can once again be born into the world, why should they miss out on the suffering?

Thanks,

Glenn

**Comment by Ascathor** [comment]
@lorac2k14 @Zyres 
Just for the sake of spreading the information, this is a nicely written definition for a hackfix:

    It's applying a temporary band aid to a large gaping wound. It's fixed for now, but it is going to cause even more problems later.
    
    An example I've recently seen: You want a person named "Jim", to always appear first in an alphabetical list. To quickly solve it, you rename him to " Jim". This is a hack that will surely come back to bite you later.

https://softwareengineering.stackexchange.com/questions/73334/what-is-a-hack

**Comment by Aokromes** [comment]
2d0e653

**Pull request closed without merging**